packages/core/strapi/lib/middlewares/session.js

```diff
@@ -0,0 +1,40 @@
+'use strict';
+
+const { defaultsDeep, isEmpty } = require('lodash/fp');
+const session = require('koa-session');
+
+const defaultConfig = {
+  client: 'cookie',
+  key: 'strapi.sid',
+  prefix: 'strapi:sess:',
+  ttl: 864000000,
+  rolling: false,
+  secretKeys: [],
+  cookie: {
+    path: '/',
+    httpOnly: true,
+    maxAge: 864000000,
+    rewrite: true,
+    signed: false,
+  },
+};
+
+module.exports = (userConfig, { strapi }) => {
+  if (!isEmpty(userConfig.client) && userConfig.client !== 'cookie') {
+    throw new Error(
+      'The session middleware only handles the `cookie` client. Please update your config.'
+    );
+  }
+
+  const config = defaultsDeep(defaultConfig, userConfig);
+  config.client = 'cookie'; // forces cookie store
+
+  strapi.server.app.keys = config.secretKeys;
```

Comment:
Should we add a check to ensure that at least one key has been provided? I haven't tested if it work without any keys, but I can't imagine it to be secure.
If not throwing an `Error`, logging a warning could go a long way, don't you think?